sky/clouds/kubernetes.py

```diff
@@ -0,0 +1,364 @@
+        )
+        match = pattern.match(name)
+        if match:
+            cpus = float(match.group('cpus'))
+            memory = float(match.group('memory'))
+            accelerator_count = match.group('accelerator_count')
+            accelerator_type = match.group('accelerator_type')
+            if accelerator_count:
+                accelerator_count = float(accelerator_count)
+                accelerator_type = str(accelerator_type)
+            else:
+                accelerator_count = None
+                accelerator_type = None
+            return cpus, memory, accelerator_count, accelerator_type
+        else:
+            raise ValueError(f'Invalid instance name: {name}')
+
+    @classmethod
+    def from_instance_type(cls, name: str) -> 'KubernetesInstanceType':
+        """Returns an instance name object from the given name."""
+        if not cls.is_valid_instance_type(name):
+            raise ValueError(f'Invalid instance name: {name}')
+        cpus, memory, accelerator_count, accelerator_type = \
+            cls._parse_instance_type(name)
+        return cls(cpus=cpus,
+                   memory=memory,
+                   accelerator_count=accelerator_count,
+                   accelerator_type=accelerator_type)
+
+    @classmethod
+    def from_resources(cls,
+                       cpus: float,
+                       memory: float,
+                       accelerator_count: float = 0,
+                       accelerator_type: str = '') -> 'KubernetesInstanceType':
+        """Returns an instance name object from the given resources."""
+        name = f'{cpus}CPU--{memory}GB'
+        if accelerator_count > 0:
+            name += f'--{accelerator_count}{accelerator_type}'
+        return cls(cpus=cpus,
+                   memory=memory,
+                   accelerator_count=accelerator_count,
+                   accelerator_type=accelerator_type)
+
+    def __str__(self):
+        return self.name
+
+
+@clouds.CLOUD_REGISTRY.register
+class Kubernetes(clouds.Cloud):
+    """Kubernetes."""
+
+    SKY_SSH_KEY_SECRET_NAME = f'sky-ssh-{common_utils.get_user_hash()}'
+
+    # TODO(romilb): Make the timeout configurable.
+    TIMEOUT = 60  # Timeout for resource provisioning
+
+    _DEFAULT_NUM_VCPUS = 2
+    _DEFAULT_MEMORY_CPU_RATIO = 1
+    _REPR = 'Kubernetes'
+    _regions: List[clouds.Region] = [clouds.Region('kubernetes')]
+    _CLOUD_UNSUPPORTED_FEATURES = {
+        # TODO(romilb): Stopping might be possible to implement with
+        #  container checkpointing introduced in Kubernetes v1.25. See:
+        #  https://kubernetes.io/blog/2022/12/05/forensic-container-checkpointing-alpha/ # pylint: disable=line-too-long
+        clouds.CloudImplementationFeatures.STOP: 'Kubernetes does not '
+                                                 'support stopping VMs.',
+        clouds.CloudImplementationFeatures.AUTOSTOP: 'Kubernetes does not '
```

Comment:
Ah this is a good point. I've filed #2249 and will create another PR for this since this requires changes (selecting autostop based on cloud being provisioned) which are slightly orthogonal to k8s support.